enterprise/server/backends/userdb/userdb.go

```diff
@@ -58,6 +58,12 @@ var (
 	}
 )
 
+type userGroupJoin struct {
+	tables.User
+	tables.UserGroup
+	tables.Group
```

Comment:
nit: would it work to make these 2 fields pointers (i.e. nil-able)? I feel like that would be slightly less error prone than checking for empty IDs in these fields (which could lead to returning wrong data), and also makes it slightly more clear that these are left-joined with User rather than inner-joined.